astropy_timeseries/binned.py

```diff
@@ -160,3 +160,88 @@ def __getitem__(self, item):
                                                  keys=self.groups._keys)
                 return out
         return super().__getitem__(item)
+
+    @classmethod
+    def read(self, filename, format, time_column=None, time_bin_start=None, time_bin_end=None,
+             time_bin_size_column=None, time_bin_size=None, time_bin_unit=None, *args, **kwargs):
+        """
+        Read and parse a file and returns a `astropy_timeseries.binned.BinnedTimeSeries`.
+
+        This function provides the access to the astropy unified I/O layer.
+        This allows easily reading a file in many supported data formats
+        using syntax such as::
+
+          >>> from astropy_timeseries.binned import BinnedTimeSeries
+          >>> dat = BinnedTimeSeries.read('table.dat', format='ascii', time_column='DATE')  # doctest: +SKIP
+
+        See also: http://docs.astropy.org/en/stable/io/unified.html
+
+        Parameters
+        ----------
+        filename: str
+            File to parse.
+        format : str
+            File format specifier.
+        time_column: str, optional
+            The name of the time column within the file.
+            This or ``time_bin_start`` must be passed in.
+        time_bin_start: str or list of str, optional
+            Can be a `astropy.time.Time` parseable string or a list of said strings.
+            This or ``time_column`` must be passed in.
+        time_bin_end: str, optional
+            The end time for the last binned data point.
+        time_bin_size: float or list, optional
+            The time step between each time index. If evenly sampled, you can specify
+            one single time step. If arbitrarily sampled, you can pass in a list.
+        time_bin_size_column: str, optional
+            If the time_bin_size is in the data file, use load it into the BinnedTimeSeries.
+        time_bin_unit: `astropy.units.quantity.Quantity`, optional
+            If you pass in a time_bin_size_column, you have to specify the unit for that column.
+            This should be a `astropy.units` time quantity.
+        *args : tuple, optional
+            Positional arguments passed through to the data reader.
+            If supplied, the first argument is the input filename.
+        **kwargs : dict, optional
+            Keyword arguments passed through to the data reader.
+
+        Returns
+        -------
+        out : `astropy_timeseries.binned.BinnedTimeSeries`
+            BinnedTimeSeries corresponding to file contents.
+
+        """
+
+        table = Table.read(filename, format=format, *args, **kwargs)
```

Comment:
Here to have access to the existing readers, I think you should first check ``super().read(...)`` and just return that unless it raises an ``IORegistryError``. This means we first check explicit formats defined for time series before falling back on the behavior below.